app/models/course/group.rb

```diff
@@ -30,10 +30,16 @@ def should_create_manager?
       !group_users.exists?(user: creator)
   end
 
-  def users_are_unique
-    new_group_users = group_users.select(&:new_record?)
-    return if new_group_users.count == new_group_users.uniq(&:user).count
-
-    errors.add(:'group_users.user_id',  I18n.t('errors.messages.taken'))
+  def validate_users_are_unique
+    users = Set.new
+    group_users.each do |group_user|
```

Comment:
Sure, this method here is only for group.valid? to return false, the errors to group_users will be added in: https://github.com/Coursemology/coursemology2/blob/master/app/controllers/course/groups_controller.rb#L49